crates/flightrepl/src/completer.rs

```diff
@@ -0,0 +1,552 @@
+    )
+    .await?;
+
+    let mut udfs = Vec::new();
+    for batch in records.0 {
+        if let Some(array) = batch.column(0).as_any().downcast_ref::<StringArray>() {
+            for value in array.iter() {
+                if let Some(column_name) = value {
+                    udfs.push(column_name.to_string());
+                }
+            }
+        }
+    }
+
+    Ok(udfs)
+}
+
+fn extract_word(line: &str, pos: usize) -> (usize, &str) {
+    let pos = pos.min(line.len());
+    let chars: Vec<char> = line.chars().collect();
+
+    // Find start of current word
+    let mut start = pos;
+    while start > 0 {
+        let ch = chars[start - 1];
+        if is_word_boundary(ch) {
+            break;
+        }
+        start -= 1;
+    }
+
+    // Find end of current word
+    let mut end = pos;
+    while end < chars.len() {
+        let ch = chars[end];
+        if is_word_boundary(ch) {
+            break;
+        }
+        end += 1;
+    }
+
+    let start_byte = chars[..start].iter().map(|c| c.len_utf8()).sum();
+    let end_byte = chars[..end].iter().map(|c| c.len_utf8()).sum();
+
+    (start, &line[start_byte..end_byte])
+}
+
+fn is_word_boundary(ch: char) -> bool {
+    match ch {
+        ' ' | '\t' | '\n' | '\r' => true,
+        '(' | ')' | ',' | ';' | '=' | '<' | '>' | '!' | '+' | '-' | '*' | '/' | '%' => true,
+        '\'' | '"' | '`' => true,
+        '.' | '[' | ']' | '{' | '}' | '|' | '&' | '^' | '~' => true,
+        _ => false,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rustyline::history::MemHistory;
+
+    fn create_test_editor_helper() -> EditorHelper {
+        let schema_cache = Arc::new(RwLock::new(SchemaCache {
+            udfs: vec!["count".to_string(), "sum".to_string(), "concat".to_string()],
+            tables: vec![
+                "users".to_string(),
+                "products".to_string(),
+                "orders".to_string(),
+                "user_profiles".to_string(),
+            ],
+            columns: vec![
+                "id".to_string(),
+                "name".to_string(),
+                "email".to_string(),
+                "age".to_string(),
+                "price".to_string(),
+                "product_name".to_string(),
+                "user_id".to_string(),
+                "order_date".to_string(),
+                "profile_picture".to_string(),
+            ],
+            keywords: ALL_KEYWORDS.iter().map(|k| k.to_lowercase()).collect(),
+        }));
+
+        EditorHelper {
+            schema_cache,
+            flight_client: None,
+            api_key: None,
+            user_agent: "test".to_string(),
+            refresh_task_handle: None,
+            shutdown_sender: None,
+        }
+    }
+
+    fn get_completions(helper: &EditorHelper, line: &str, pos: usize) -> Vec<String> {
+        let history = MemHistory::new();
+        let ctx = Context::new(&history);
+        let result = helper.complete(line, pos, &ctx).unwrap();
```

Comment:
```suggestion
        let result = helper.complete(line, pos, &ctx).expect("Should complete suggestion");
```

Prefer `.expect()` instead of `.unwrap()`

Reply:
Thanks, done